driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -104,61 +107,57 @@ public void open() throws IOException {
     @SuppressWarnings("deprecation")
     @Override
     public void openAsync(final AsyncCompletionHandler<Void> handler) {
-        Bootstrap bootstrap = new Bootstrap();
-        bootstrap.group(workerGroup);
-        bootstrap.channel(socketChannelClass);
-
-        bootstrap.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, settings.getConnectTimeout(MILLISECONDS));
-        bootstrap.option(ChannelOption.TCP_NODELAY, true);
-        bootstrap.option(ChannelOption.SO_KEEPALIVE, settings.isKeepAlive());
-
-        if (settings.getReceiveBufferSize() > 0) {
-            bootstrap.option(ChannelOption.SO_RCVBUF, settings.getReceiveBufferSize());
-        }
-        if (settings.getSendBufferSize() > 0) {
-            bootstrap.option(ChannelOption.SO_SNDBUF, settings.getSendBufferSize());
-        }
-        bootstrap.option(ChannelOption.ALLOCATOR, allocator);
+        final Queue<SocketAddress> socketAddressQueue = new LinkedList<SocketAddress>(address.getSocketAddresses());
+        initializeChannel(handler, socketAddressQueue);
+    }
 
-        bootstrap.handler(new ChannelInitializer<SocketChannel>() {
-            @Override
-            public void initChannel(final SocketChannel ch) throws Exception {
-                if (sslSettings.isEnabled()) {
-                    SSLEngine engine = getSslContext().createSSLEngine(address.getHost(), address.getPort());
-                    engine.setUseClientMode(true);
-                    SSLParameters sslParameters = engine.getSSLParameters();
-                    enableSni(address.getHost(), sslParameters);
-                    if (!sslSettings.isInvalidHostNameAllowed()) {
-                        enableHostNameVerification(sslParameters);
-                    }
-                    engine.setSSLParameters(sslParameters);
-                    ch.pipeline().addFirst("ssl", new SslHandler(engine, false));
-                }
-                int readTimeout = settings.getReadTimeout(MILLISECONDS);
-                if (readTimeout > 0) {
-                    ch.pipeline().addLast(READ_HANDLER_NAME, new ReadTimeoutHandler(readTimeout));
-                }
-                ch.pipeline().addLast(new InboundBufferHandler());
+    @SuppressWarnings("deprecation")
+    public void initializeChannel(final AsyncCompletionHandler<Void> handler, final Queue<SocketAddress> socketAddressQueue) {
+        if (socketAddressQueue.isEmpty()) {
```

Comment:
Can this ever be true?

Reply:
The only time it has a chance of being true is when it is called the first time. If the hostname is just completely incorrect and doesn't have any associated ip addresses, then the queue will be empty when it is initialized. Is this something that could happen?

Reply:
Ack - lets keep it in, I'm not 100% sure if it could be empty but as its async its best kept left in so the handler is called.